Source/WebCore/css/SelectorChecker.cpp

```diff
@@ -269,7 +269,7 @@ static SelectorChecker::LocalContext localContextForParent(const SelectorChecker
     }
 
     // Move to the shadow host if matching :host and the parent is the shadow root.
-    if (context.selector->match() == CSSSelector::Match::PseudoClass && context.selector->pseudoClass() == CSSSelector::PseudoClass::Host) {
+    if (context.selector->isOrSimilarToHostPseudoClass()) {
```

Comment:
I still don't think this is right conceptually as `:is()` may contain other things besides `:host` we don't know if we got here by matching a `:host` or not. We should move to the host scope when we try to match`:host` inside.

Maybe in

```
        case CSSSelector::PseudoClass::Is:
        case CSSSelector::PseudoClass::Where:
        case CSSSelector::PseudoClass::WebKitAny:
```

branch we could move to the host element similar to localContextForParent() if we see `PseudoElement::Host` as the rightmost component of the subselector.

Reply:
Good point. I've changed the fix.